doc/tokens.md

```diff
@@ -1,148 +1,251 @@
+JWT authentication works with BOSH and WebSocket connections.
+
+### Token Structure
 
 The following JWT claims are used in the authentication token:
-- 'iss' specifies the *application ID* which identifies the client app connecting to the server. It should be negotiated with the service provider before generating the token.
-- 'room' contains the name of the room for which the token has been allocated. This is *NOT* the full MUC room address. An example assuming that we have full MUC 'conference1@muc.server.net' would be that 'conference1' should be used here.  Alternately, a '*' may be provided, allowing access to all rooms within the domain.
-- 'exp' token expiration timestamp as defined in the RFC
-- 'sub' contains EITHER the lowercase name of the tenant (for a conference like TENANT1/ROOM with would be 'tenant1') OR the lowercase name of the domain used when authenticating with this token (for a conference like /ROOM). By default assuming that we have full MUC 'conference1@muc.server.net' then 'server.net' should be used here.  Alternately, a '*' may be provided, allowing access to rooms in all tenants within the domain or all domains within the server.
-- 'aud' application identifier. This value indicates what service is consuming the token.  It should be negotiated with the service provider before generating the token.
 
-The secret is used to compute the HMAC hash value and verify the token for HS256 tokens.  
-- It is now possible to define the algorithm type used, simply update the prosody.cfg.lua file with your chosen type. e.g `signature_algorithm = "HS512"`
+- 'iss'
+  - specifies the _application ID_ which identifies the client.
+    It should be negotiated with the service provider before generating the token.
+- 'room'
+  - contains either
+    - the name of the room - This is *NOT* the full MUC room address.  
+      For a full MUC like 'conference1@muc.server.net' it would be 'conference1'.
+    - alternatively '*' - allowing access to all rooms within the domain.
+- 'exp'
+  - token expiration timestamp as defined in the [RFC7519]
+- 'sub'
+  - contains either
+    - the lowercase _tenant_ - for a conference like 'TENANT1/ROOM' it would be 'tenant1', see example below.
+    - the lowercase _domain_ - for a full MUC like 'conference1@muc.server.net' it would be 'server.net'.
+    - alternatively '*' - allowing access to all rooms in all tenants within the domain or all domains within the server.
+- 'aud'
+  - application identifier - indicates what service is consuming the token.
+    It should be negotiated with the service provider before generating the token.
+
+**Sub with tenant** If the 'sub' contains the tenant. You need to define the tenant in the `muc` when creating a new `JitsiConnection` in the options 
+within the `hosts` section, see
+
+```javascript
+    const connection = new JitsiMeetJS.JitsiConnection(
+    '${applicationId}',
+    '${token}',
+    {
+        serviceUrl: 'wss://muc.server.net/xmpp-websocket',
```

Comment:
In other places `muc.server.net` is referenced as the muc component. So here the websocket needs to have the address `server.net`.
You also need to add the tenant here to the address. So it would be `'wss://server.net/tenant/xmpp-websocket'`
And it uses this nginx rule to set some correct params: https://github.com/jitsi/jitsi-meet/blob/fcb1845e2a6d7778dacd29f5c75e3d48c52b9c47/doc/debian/jitsi-meet/jitsi-meet.example#L158

Reply:
done.